Scripts/Utilities/TransitionUtils.cs

```diff
@@ -0,0 +1,71 @@
+﻿#if UNITY_EDITOR
+using System;
+using System.Reflection;
+using UnityEditor.Experimental.EditorVR.Modules;
+using UnityEditor.Experimental.EditorVR.UI;
+using UnityEngine;
+
+namespace UnityEditor.Experimental.EditorVR.Utilities
+{
+    /// <summary>
+    /// Transition utilities
+    /// </summary>
+    public static class TransitionUtils
+    {
+        public static void AnimateProperty<T>(float time, bool state, ref bool lastState, ref float changeTime,
```

Comment:
Do we want some standard for organizing args?  Like normal, ref, out, function (or some other order?)

Reply:
Ideally, yeah.  IMO there should be a standard for the core args.

Reply:
I'm not sure I agree that ref/out/standard is the criteria for argument order. Obviously default parameters have to go last as required, but otherwise I try to group them conceptually. In this case, I actually switched back and forth between ref as I was writing the method, and I can imagine that future refactors might turn certain arguments into or out of being ref.